mmc-google-analytics/src/main/java/info/novatec/metricscollector/google/MetricsRepository.java

```diff
@@ -25,21 +25,24 @@
     private final MetricsValidator metricsValidator;
 
     public void saveMetrics(List<Metrics> metricsForAllPages, String measurementName) {
-        List<Point> points = new ArrayList<>();
-        metricsForAllPages.forEach(metrics -> points.addAll(createPoints(metrics, measurementName)));
-        log.info("Created {} points for measurement '{}'.", points.size(), measurementName);
-        influx.savePoint(points);
+        influx.savePoint(createPoints(metricsForAllPages, measurementName));
         influx.close();
     }
 
-    private List<Point> createPoints(Metrics metrics, String measurementName) {
+    List<Point> createPoints(List<Metrics> metricsForAllPages, String measurementName) {
         List<Point> points = new ArrayList<>();
+        metricsForAllPages.forEach(metrics -> points.add(createPointBuilder(metrics, measurementName).build()));
+        log.info("Created {} points for measurement '{}'.", points.size(), measurementName);
+        return points;
+    }
 
+    Point.Builder createPointBuilder(Metrics metrics, String measurementName) {
+        log.info("Start creating points for page '{}'.", metrics.getPagePath());
         if (metricsValidator.hasNullValues(metrics)) {
             log.error(
                 "Since there are null values in metrics, creating points for page '{}' isn't possible! Metrics content:\n{}",
                 metrics.getPagePath(), metrics.toString());
-            return points;
+            return null;
```

Comment:
Is returning null safe?

Reply:
Good point. I dont know, what influx does with a points-list that has null-values. I filtered them out.